lib/src/main/java/graphql/nadel/NadelSchemas.kt

```diff
@@ -0,0 +1,177 @@
+            schemaTransformationHook = value
+        }
+
+        fun overallWiringFactory(value: WiringFactory): Builder = also {
+            overallWiringFactory = value
+        }
+
+        fun underlyingWiringFactory(value: WiringFactory): Builder = also {
+            underlyingWiringFactory = value
+        }
+
+        fun serviceExecutionFactory(value: ServiceExecutionFactory): Builder = also {
+            serviceExecutionFactory = value
+        }
+
+        fun overallSchema(serviceName: String, schema: Reader): Builder = also {
+            overallSchemas[serviceName] = schema
+        }
+
+        fun underlyingSchema(serviceName: String, schema: Reader): Builder = also {
+            underlyingSchemas[serviceName] = schema
+        }
+
+        fun overallSchema(serviceName: String, schema: String): Builder = also {
+            overallSchemas[serviceName] = schema.reader()
+        }
+
+        fun underlyingSchema(serviceName: String, schema: String): Builder = also {
+            underlyingSchemas[serviceName] = schema.reader()
+        }
+
+        @JvmName("overallSchemasReader")
+        fun overallSchemas(value: Map<String, Reader>): Builder = also {
+            overallSchemas = value.toMutableMap() // copy
+        }
+
+        @JvmName("underlyingSchemasReader")
+        fun underlyingSchemas(value: Map<String, Reader>): Builder = also {
+            underlyingSchemas = value.toMutableMap() // copy
+        }
+
+        @JvmName("overallSchemasString")
+        fun overallSchemas(value: Map<String, String>): Builder = also {
+            overallSchemas = value
+                .mapValuesTo(LinkedHashMap()) { (_, schema) ->
+                    StringReader(schema)
+                }
+        }
+
+        @JvmName("underlyingSchemasString")
+        fun underlyingSchemas(value: Map<String, String>): Builder = also {
+            underlyingSchemas = value
+                .mapValuesTo(LinkedHashMap()) { (_, schema) ->
+                    StringReader(schema)
+                }
+        }
+
+        /**
+         * Use this if you want just want the service schema without the "full" [Service].
+         */
+        fun stubServiceExecution(): Builder = also {
+            serviceExecutionFactory(
+                object : ServiceExecutionFactory {
+                    override fun getServiceExecution(serviceName: String): ServiceExecution {
+                        return ServiceExecution {
+                            throw UnsupportedOperationException("no-op")
+                        }
+                    }
+                },
+            )
+        }
+
+        fun build(): NadelSchemas {
+            require(overallSchemas.isNotEmpty()) { "Nadel schemas must not be empty" }
+            require(underlyingSchemas.isNotEmpty()) { "Underlying schemas must not be empty" }
+
+            require(underlyingSchemas.keys.containsAll(overallSchemas.keys)) {
```

Comment:
thinking about edge cases which will likely never happen :)
`containsAll` will return true when `underlyingSchemas.keys` is a superset of `overallSchemas.keys`, should it be `equals` instead?

Reply:
Yeah I deliberately wrote it like that, but now that I'm thinking about it again, yea maybe being defensive here is better… could show a mistake was made.

Reply:
@temaEmelyan done